packages/rocketchat-language-plugin/server/leave_automatic_channel.js

```diff
@@ -0,0 +1,18 @@
+Meteor.methods({
+	leave_automatic_channel(room_name) {
+		const user_id = this.userId;
+		const user_info = RocketChat.models.Users.findOneById(user_id);
+		const user_info_language_channels = user_info.automatic_channels[0].language.added_to;
+		const blacklisted = user_info.automatic_channels[0].language.blacklisted;
+		if (user_info_language_channels.includes(room_name)) {
+			const index = user_info_language_channels.indexOf(room_name);
+			if (index > -1) {
+				user_info_language_channels.splice(index, 1);
+				blacklisted.push(room_name);
+
+			}
+		}
+		const update_automatic_channels = {automatic_channels : user_info.automatic_channels};
+		RocketChat.models.Users.update({_id:this.userId}, {$set :update_automatic_channels});
```

Comment:
You should be able to use MongoDB queries to remove an entry directly from the database. No need to read array, remove it in JavaScript, and set it back.